src/main/java/io/vertx/core/http/impl/HttpServerImpl.java

```diff
@@ -1019,5 +1065,13 @@ private void http1(ChannelHandlerContext ctx, ByteBuf buf) {
       ctx.fireChannelRead(buf);
       ctx.pipeline().remove(this);
     }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+      Channel channel = ctx.channel();
+      HandlerHolder<HttpHandler> reqHandler = reqHandlerManager.chooseHandler(channel.eventLoop());
+      reqHandler.context.executeFromIO(channel::close);
```

Comment:
actually channel.close should be called normally but it's the callback to connectionExceptionHandler that should be executed with executeFromIO.